app/experimenter/experiments/forms.py

```diff
@@ -570,13 +585,50 @@ class ExperimentObjectivesForm(ChangeLogMixin, forms.ModelForm):
         widget=forms.Textarea(attrs={"class": "form-control", "rows": 20}),
     )
 
+    survey = forms.TypedChoiceField(
+        label=Experiment.SURVEY_REQUIRED_LABEL,
+        help_text=Experiment.SURVEY_HELP_TEXT,
+        choices=RADIO_OPTIONS,
+        widget=RadioWidgetCloser,
+        coerce=coerce_truthy,
+        empty_value=None,
+    )
+    survey_urls = forms.CharField(
+        required=False,
+        help_text=Experiment.SURVEY_HELP_TEXT,
+        label="Survey URLs",
+        widget=forms.Textarea(attrs={"class": "form-control", "rows": 1}),
+    )
+    survey_instructions = forms.CharField(
+        required=False,
+        label=Experiment.SURVEY_INSTRUCTIONS_LABEL,
+        help_text=Experiment.SURVEY_LAUNCH_INSTRUCTIONS_HELP_TEXT,
+        widget=forms.Textarea(attrs={"class": "form-control", "rows": 10}),
+    )
+
     class Meta:
         model = Experiment
-        fields = ("objectives", "analysis_owner", "analysis")
+        fields = (
+            "objectives",
+            "analysis_owner",
+            "analysis",
+            "survey",
+            "survey_urls",
+            "survey_instructions",
+        )
 
+    def clean(self):
+        # if the survey radio button is false, we make sure to not
+        # save any information for urls or instructions. we also want
+        # to make sure that if they once answered yes, and saved the
+        # form, but now are removing the survey, that we clear out the
+        # database of the url and instructions field
+        cleaned_data = super().clean()
+        if not cleaned_data.get("survey"):
+            cleaned_data["survey_urls"] = ""
+            cleaned_data["survey_instructions"] = ""
 
-class RadioWidget(forms.widgets.RadioSelect):
-    template_name = "experiments/radio_widget.html"
+        return cleaned_data
```

Comment:
Is the clean method the best place for making sure that if the user goes from Yes to No... that we are removing the urls and instructions from the DB?

Reply:
I'm not sure about architecture but it seems OK to me.